src/Microsoft.Graph.Core/Authentication/TokenCredentials/IntegratedWindowsTokenCredential.cs

```diff
@@ -0,0 +1,155 @@
+        /// </summary>
+        /// <param name="requestContext">The <see cref="TokenRequestContext"/> for the request</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> for the request</param>
+        /// <returns>An <see cref="AccessToken"/> to make requests with</returns>
+        public override async ValueTask<AccessToken> GetTokenAsync(TokenRequestContext requestContext, CancellationToken cancellationToken)
+        {
+            AuthenticationResult result;
+            try
+            {
+                // Try to get the account from the cache
+                IEnumerable<IAccount> accounts = await _clientApplication.GetAccountsAsync();
+                IAccount account = accounts.FirstOrDefault();
+
+                // Try to get the token silently
+                AcquireTokenSilentParameterBuilder tokenSilentBuilder =
+                    _clientApplication.AcquireTokenSilent(requestContext.Scopes, account);
+
+                if (!ContainsWellKnownTenantName(_clientApplication.Authority))
+                    tokenSilentBuilder.WithAuthority(_clientApplication.Authority);
+
+                result = await tokenSilentBuilder.ExecuteAsync(cancellationToken);
+            }
+            catch (MsalException)
+            {
+                // We can't get the token silently so get a new one
+                result = await GetNewAccessTokenAsync(requestContext);
+            }
+
+            return new AccessToken(result.AccessToken, result.ExpiresOn);
+        }
+
+        /// <summary>
+        /// Gets an new <see cref="AccessToken"/> after checking the cache has failed
+        /// </summary>
+        /// <param name="requestContext">The <see cref="TokenRequestContext"/> for the request</param>
+        /// <returns></returns>
+        private async Task<AuthenticationResult> GetNewAccessTokenAsync(TokenRequestContext requestContext)
+        {
+            AuthenticationResult authenticationResult = null;
+            int retryCount = 0;
+            do
+            {
+                try
+                {
+                    authenticationResult = await _clientApplication.AcquireTokenByIntegratedWindowsAuth(requestContext.Scopes)
+                                .ExecuteAsync();
+                }
+                catch (MsalServiceException serviceException)
+                {
+                    // Service not available so wait 
+                    if (serviceException.ErrorCode == ErrorConstants.Codes.TemporarilyUnavailable)
+                    {
+                        TimeSpan delay = this.GetRetryAfter(serviceException);
+                        retryCount++;
+                        // pause execution
+                        await Task.Delay(delay);
+                    }
+                    else
+                    {
+                        throw new AuthenticationException(
```

Comment:
Document this exception at the first public method where this can be encountered